### PR TITLE
Feat: default strategy UI

### PR DESCRIPTION
**Pull request by andreas-unleash**
- Removed `strategyTitle` and `strategyDisable` flags.  Unified under `strategyImprovements` flag
- Implements the default strategy UI
- Bug fixes

## About the changes

Closes # [1-875](https://linear.app/unleash/issue/1-875/default-strategy-frontend)

Screenshot 2023-05-04 at 11 21 05

### CODE REVIEW

File: src/server-dev.ts [hunk 1]
```diff
@@ -38,6 +38,9 @@ process.nextTick(async () => {
                         embedProxyFrontend: true,
                         anonymiseEventLog: false,
                         responseTimeWithAppNameKillSwitch: false,
+                        optimal304: true,
```
Comment (sjaanus) [inline src/server-dev.ts:41]: These I think were removed. Try to align with main branch.

File: frontend/src/component/project/Project/ProjectSettings/ProjectDefaultStrategySettings/ProjectEnvironment/ProjectEnvironmentDefaultStrategy/ProjectEnvironmentDefaultStrategy.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        environmentId
+    );
+
+    const path = `/projects/${projectId}/settings/default-strategy`;
+
+    const strategy: CreateFeatureStrategySchema = useMemo(() => {
+        return defaultStrategy ? defaultStrategy : DEFAULT_STRATEGY;
+    }, [JSON.stringify(defaultStrategy)]);
+
+    console.log(strategy);
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectSettings/ProjectDefaultStrategySettings/ProjectEnvironment/ProjectEnvironmentDefaultStrategy/ProjectEnvironmentDefaultStrategy.tsx:59]: Seems a leftover.

**Comment by sjaanus** [comment]
Segments seem to be missing from default strategy.

**Review by sjaanus — CHANGES_REQUESTED** [review]
Save strategy button seems to not do anything 
[Screencast from 05.05.2023 11:05:17.webm](https://user-images.githubusercontent.com/964450/236406709-b1a44010-ba29-44d1-b832-aefa99d4aa88.webm)

File: frontend/src/component/project/Project/ProjectSettings/ProjectDefaultStrategySettings/ProjectEnvironment/ProjectEnvironmentDefaultStrategy/ProjectEnvironmentDefaultStrategy.tsx (hunk 2 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectSettings/ProjectDefaultStrategySettings/ProjectEnvironment/ProjectEnvironmentDefaultStrategy/ProjectEnvironmentDefaultStrategy.tsx]: yup, thanks

**Comment by sjaanus** [comment]
[Screencast from 05.05.2023 12:53:44.webm](https://user-images.githubusercontent.com/964450/236428073-8e6f7127-6767-42c5-b63c-5f31156b493e.webm)

When editing, the strategy screen is always 100% and not using the exising settings.

**Comment by sjaanus** [comment]
If the group is defined in default strategy, it should use it instead of feature name.

**Review by sjaanus — APPROVED** [review]
Giving soft approval. If the issues get fixed, should be fine.

**Pull request merged**